lib/tasks/npm-task.js

```diff
@@ -28,7 +36,27 @@ class NpmTask extends Task {
     return RSVP.resolve(execa('npm', args, { preferLocal: false }));
   }
 
-  checkVersion() {
+  yarn(args) {
+    logger.info('yarn: %j', args);
+    return RSVP.resolve(execa('yarn', args, { preferLocal: false }));
```

Comment:
the same is true for the `npm()` method that was previously introduced. have you been able to make `execa()` throw an exception? it seems like execa knows about that contract and converts potential exceptions to Promise rejections itself (see e.g. https://github.com/sindresorhus/execa/blob/master/index.js#L128-L133)